java-checks/src/main/java/org/sonar/java/checks/BooleanLiteralCheck.java

```diff
@@ -59,10 +59,15 @@ public void visitNode(Tree tree) {
 
   @Nullable
   private static LiteralTree getBooleanLiteral(Tree... trees) {
-    return Arrays.stream(trees)
-      .filter(tree -> tree.is(Kind.BOOLEAN_LITERAL))
-      .map(LiteralTree.class::cast)
-      .findFirst().orElse(null);
+    LiteralTree booleanLiteral = null;
+    for (Tree t : trees) {
+      if (t.is(Kind.NULL_LITERAL)) {
+        return null;
+      } else if (t.is(Kind.BOOLEAN_LITERAL)) {
+        booleanLiteral = (LiteralTree) t;
```

Comment:
Reassigning `booleanLiteral` here, changes the location of the issue when there are multiple literals.

PS: Add `[[sc=X;ec=Y]]` to the tests to catch this.